File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/visitor/Visitor.java
package jv.com.tencent.cain.design_pattern.visitor;

/**
 * @author : jiangyu
 * @date : 2020/8/22
 * @desc : xxx
 */
abstract class Visitor {
    protected String name;

    public String getName() {
        return name;
    }


    public Visitor(String name) {
        this.name = name;
    }

    abstract void visitConcrectElementA(ConcrectElementA concrectElementA);

    abstract void visitConcrectElementB(ConcrectElementB concrectElementB);
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/visitor/Visitor.kt
package kt.com.tencent.cain.desigin_pattern.visitor

/**
 * @author : jiangyu
 * @date   : 2020/8/23
 * @desc   : xxx
 */
abstract class Visitor(val name:String) {
    abstract fun visitorConcrectC(concrectElementC: ConcrectElementC)
    abstract fun visitorConcrectD(concrectElementD: ConcrectElementD)
}